docroot/modules/custom/uiowa_core/uiowa_core.module

```diff
@@ -1092,6 +1092,7 @@ function uiowa_core_set_field_icon(array &$variables, string $icon_classes) : vo
  *   The class(es) to be added to the block/field's attributes.
  */
 function uiowa_core_set_attributes(array &$variables, array|string $classes) : void {
+  $variables['attributes']['class'] = [];
```

Comment:
This seems like it contradicts the section right below it.

Reply:
I was getting `TypeError: array_merge(): Argument #1 must be of type array, string given in array_merge() ` error on https://github.com/uiowa/uiowa/blob/bd01e3694fc7fb1f9286787569d096c986183360/docroot/sites/admissions.uiowa.edu/modules/admissions_core/admissions_core.module#L382.  Any suggestion on how to fix that?

Reply:
It looks like `$variables['attributes']['class']` in some cases is just a string, while in others it's an array of classes. 

One fix would be to handle it similar to the check for `$classes`, after checking if it's set, like
```
if (!is_array($variables['attributes']['class']) {
  $variables['attributes']['class'] = [$variables['attributes']['class']];
}
```
Otherwise, we could split it to be an `array_merge` in the case that it's an array, and otherwise add the string to the `$classes` array, then assign that to the `$variables['attributes']['class']`

Reply:
It would appear that the `$variables['attributes']['class']` was previously set to a string, so I would look at making sure that it is an array before it gets passed into this function.

Reply:
I'm not sure about the approach of converting a string to an array in this case. Instead of it being an intentional choice, it is a catch for cases where we don't understand why we are getting a string in the first place. It sort of hides the problem that we should be figuring out upstream of this.

Reply:
I assume it's the same or similar to https://github.com/uiowa/uiowa/issues/5577, and that we may run into it in several places, but this could be good motivation to do some cleanup.

Reply:
And to maintain consistency, we may want to pass multiple classes in as an array of classes when calling, like `uiowa_core_set_attributes($variables, ['text-align-center', 'headline headline--serif', 'headline--highlight');`, and use a string only if its a single class.

Reply:
This would be my preference.

Reply:
Thanks all, I will update based on @cory-skeers's suggestion.